File: Paint_BackEnd/src/main/java/com/paint/Paint_BackEnd/Shapes/Rectangle.java
package com.paint.paint_backend.Shapes;

public class Rectangle extends IShape{
    private int _width;
    private int _height;

    Rectangle(int x, int y, String stroke, double strokeWidth, String fill, int rotate, boolean draggable, String id, boolean update, boolean deleteflag, int width, int height){
        super(x, y, stroke, strokeWidth, fill, rotate, draggable, id, update, deleteflag);
        this._width = width;
        this._height = height;
        super.set_type("rectangle");
    }

    public int get_width() {
        return _width;
    }

    public void set_width(int _width) {
        this._width = _width;
    }

    public int get_height() {
        return _height;
    }

    public void set_height(int _height) {
        this._height = _height;
    }

}

File: Paint_FrontEnd/src/app/Shapes/rectangle.ts
import { IShape } from "./ishape";

export class Rectangle extends IShape{
    private _width: number;
    private _height: number;

    constructor(x: number, y: number, stroke: string, strokeWidth: number, fill: string, rotate: number, draggable: boolean, id: string, update: boolean, deleteflag: boolean, width: number, height: number){
        super(x, y, stroke, strokeWidth, fill, rotate, draggable, id, update, deleteflag);
        this._width = width;
        this._height = height;
        this.type = "rectangle";
    }

    public get width(): number {
        return this._width;
    }
    public set width(value: number) {
        this._width = value;
    }
    public get height(): number {
        return this._height;
    }
    public set height(value: number) {
        this._height = value;
    }
}
